lib/bridge/MatrixHandler.js

```diff
@@ -468,11 +468,53 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
         return;
     }
     else if (cmd === "!quit") {
-        // clientList.filter(
-        //     (bridgedClient) => bridgedClient.server.domain === ircServer.domain
-        // ).forEach((bridgedClient) => {
-        //     bridgedClient.kill();
-        // });
+        // Filter to get the clients for the [specified] server
+        let clients = clientList.filter(
+            (bridgedClient) => bridgedClient.server.domain === ircServer.domain
+        );
+        if (clients.length === 0) {
+            req.log.info(`No bridgedClients for ${event.user_id}`);
+            let notice = new MatrixAction("notice", "You are not connected to any networks.");
+            yield this.ircBridge.sendMatrixAction(adminRoom, botUser, notice, req);
+        }
+        let getRoomsForChannel = (channel) => {
+            return this.ircBridge.getStore().getMatrixRoomsForChannel(ircServer, channel);
+        };
+        for (var i = 0; i < clients.length; i++) {
+            let bridgedClient = clients[i];
+            // Get all rooms that the bridgedClient is in
+            let rooms = yield Promise.all(bridgedClient.chanList.map(getRoomsForChannel));
+
+            // rooms is an array of arrays
+            rooms = rooms.reduce((a, b) => {return a.concat(b)});
```

Comment:
Err, no it's not? https://github.com/matrix-org/matrix-appservice-irc/blob/master/lib/DataStore.js#L240